tests/pipeline/test_engine.py

```diff
@@ -1056,7 +1056,12 @@ def init_class_fixtures(cls):
             index=dates,
             columns=cls.asset_finder.retrieve_all(sids),
         )
+        cls.raw_data_with_nans = cls.raw_data.where(cls.raw_data % 3 != 0)
```

Comment:
Actually, I would also use `% 2` here, because otherwise this data is just all NaNs for the first asset and all filled values for the others. This doesn't test the case of having some NaNs and some values.